crates/ty_python_semantic/src/types/known_instance.rs

```diff
@@ -322,6 +322,12 @@ impl<'db> KnownInstanceType<'db> {
             | Self::TypeVar(_) => {
                 matches!(module, KnownModule::Typing | KnownModule::TypingExtensions)
             }
+            Self::TypingSelf => {
+                matches!(
+                    module,
+                    KnownModule::Typing | KnownModule::TypingExtensions | KnownModule::Typeshed
+                )
+            }
```

Comment:
@carljm, it looks like this change came from one of your commits. What's the motivation for this change? `_typeshed.Self` is just a legacy TypeVar -- it's very different to `typing(_extensions).Self`. You can see it defined here:

https://github.com/astral-sh/ruff/blob/04dc48e17c2518f97436d76284a509499087d81c/crates/ty_vendored/vendor/typeshed/stdlib/_typeshed/__init__.pyi#L36-L40

The only reason for its existence is that PEP 673 (somewhat inexplicably) has an arbitrary ban on using `typing(_extensions).Self` in any metaclass methods

Reply:
https://github.com/astral-sh/ruff/pull/18377